From c1e685d4039bcaa39013852db11a08711928fa38 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Tue, 23 Nov 2021 23:06:15 -0800
Subject: [PATCH 3323/4585] [Huawei] sched: sysctl_blocked_averages

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 185797

--------------------------------

Since update_blocked_averages is not include in
CONFIG_SMP, sysctl handler sysctl_blocked_averages
should not include in CONFIG_SMP as well.

Signed-off-by: Yu Jiahua <yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>

Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=20644
Signed-off-by: Zheng Zucheng <zhengzucheng@huawei.com>
---
 kernel/sched/fair.c | 68 ++++++++++++++++++++++-----------------------
 1 file changed, 34 insertions(+), 34 deletions(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 40b30a7e40fc..f651ae268d9b 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -899,6 +899,39 @@ static void update_tg_load_avg(struct cfs_rq *cfs_rq)
 }
 #endif /* CONFIG_SMP */
 
+#ifdef CONFIG_IAS_SMART_LOAD_TRACKING
+DEFINE_STATIC_KEY_TRUE(sched_blocked_averages);
+
+static void set_blocked_averages(bool enabled)
+{
+	if (enabled)
+		static_branch_enable(&sched_blocked_averages);
+	else
+		static_branch_disable(&sched_blocked_averages);
+}
+
+int sysctl_blocked_averages(struct ctl_table *table, int write,
+			    void __user *buffer, size_t *lenp, loff_t *ppos)
+{
+	struct ctl_table t;
+	int err;
+	int state = static_branch_likely(&sched_blocked_averages);
+
+	if (write && !capable(CAP_SYS_ADMIN))
+		return -EPERM;
+
+	t = *table;
+	t.data = &state;
+	err = proc_dointvec_minmax(&t, write, buffer, lenp, ppos);
+	if (err)
+		return err;
+	if (write)
+		set_blocked_averages(state);
+
+	return err;
+}
+#endif
+
 /*
  * Update the current task's runtime statistics.
  */
@@ -4723,7 +4756,7 @@ int sysctl_tick_update_load(struct ctl_table *table, int write,
 	t = *table;
 	t.data = &state;
 	err = proc_dointvec_minmax(&t, write, buffer, lenp, ppos);
-	if (err < 0)
+	if (err)
 		return err;
 	if (write)
 		set_tick_update_load(state);
@@ -8738,39 +8771,6 @@ static void attach_tasks(struct lb_env *env)
 	rq_unlock(env->dst_rq, &rf);
 }
 
-#ifdef CONFIG_IAS_SMART_LOAD_TRACKING
-DEFINE_STATIC_KEY_TRUE(sched_blocked_averages);
-
-static void set_blocked_averages(bool enabled)
-{
-	if (enabled)
-		static_branch_enable(&sched_blocked_averages);
-	else
-		static_branch_disable(&sched_blocked_averages);
-}
-
-int sysctl_blocked_averages(struct ctl_table *table, int write,
-			    void __user *buffer, size_t *lenp, loff_t *ppos)
-{
-	struct ctl_table t;
-	int err;
-	int state = static_branch_likely(&sched_blocked_averages);
-
-	if (write && !capable(CAP_SYS_ADMIN))
-		return -EPERM;
-
-	t = *table;
-	t.data = &state;
-	err = proc_dointvec_minmax(&t, write, buffer, lenp, ppos);
-	if (err < 0)
-		return err;
-	if (write)
-		set_blocked_averages(state);
-
-	return err;
-}
-#endif
-
 #ifdef CONFIG_NO_HZ_COMMON
 static inline bool cfs_rq_has_blocked(struct cfs_rq *cfs_rq)
 {
-- 
2.17.1

